src/cloudforet/cost_analysis/manager/job_manager.py

```diff
@@ -0,0 +1,113 @@
+import logging
+import re
+from datetime import datetime, timedelta
+
+from spaceone.core.manager import BaseManager
+from cloudforet.cost_analysis.error import *
+from cloudforet.cost_analysis.conf.cost_conf import *
+from cloudforet.cost_analysis.connector import BigqueryConnector, CloudBillingConnector
+from cloudforet.cost_analysis.model import Tasks
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class JobManager(BaseManager):
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.bigquery_connector: BigqueryConnector = self.locator.get_connector(BigqueryConnector)
+        self.cloud_billing_connector: CloudBillingConnector = self.locator.get_connector(CloudBillingConnector)
+        self.billing_account_id = None
+
+    def get_tasks(self, options, secret_data, schema, start, last_synchronized_at, domain_id):
+
+        tasks = []
+        changed = []
+
+        start_time = self._get_start_time(start, last_synchronized_at)
+        start_date = start_time.strftime('%Y-%m-%d')
+        changed_time = start_time
+
+        self.bigquery_connector.create_session(options, secret_data, schema)
+        self.cloud_billing_connector.create_session(options, secret_data, schema)
+
+        billing_dataset = self._get_billing_dataset_from_secret_data(secret_data)
+        billing_info = self.cloud_billing_connector.get_billing_info()
+        prefix, self.billing_account_id = billing_info['billingAccountName'].split('/')
+        target_project_ids = self._get_target_project_ids(secret_data['target_project_id'])
+
+        secret_type = options.get('secret_type', SECRET_TYPE_DEFAULT)
+
+        if secret_type == 'MANUAL':
+            # NOT IMPLEMENTED
+            pass
+        elif secret_type == 'USE_SERVICE_ACCOUNT_SECRET':
+
+            if target_project_ids:
+                for target_project_id in target_project_ids:
+                    task = {
+                        'task_options': {
+                            'start': start_date,
+                            'billing_dataset': billing_dataset,
+                            'sub_billing_account': self.billing_account_id,
+                            'target_project_id': target_project_id
+                        }
+                    }
+
+                    tasks.append(task)
+                    changed.append({'start': changed_time})
+            else:
+                task = {
+                    'task_options': {
+                        'start': start_date,
+                        'billing_dataset': billing_dataset,
+                        'sub_billing_account': self.billing_account_id,
+                        'target_project_id': '*'
+                    }
+                }
+
+                tasks.append(task)
+                changed.append({'start': changed_time})
+        else:
+            raise ERROR_INVALID_SECRET_TYPE(secret_type=options.get('secret_type'))
```

Comment:
I think this part is so long and duplicated, I would rather change it to like the one below such as call the '_generate_task_option` method.

```python
        if secret_type == "MANUAL":
            # NOT IMPLEMENTED
            pass
        elif secret_type == "USE_SERVICE_ACCOUNT_SECRET":
            if target_project_ids:
                for target_project_id in target_project_ids:
                    task.append(
                        self._generate_task_option(
                            start_date, billing_dataset, target_project_id
                        )
                    )
                    changed.append({"start": changed_time})
            else:
                task.append(
                    self._generate_task_option(start_date, billing_dataset, "*")
                )
        else:
            raise ERROR_INVALID_SECRET_TYPE(secret_type=options.get('secret_type'))      
```

And, make another method like the one below.

```python
    def _generate_task_option(self, start_date, billing_dataset, target_project_id):
        return {
            "task_options": {
                "start": start_date,
                "billing_dataset": billing_dataset,
                "sub_billing_account": self.billing_account_id,
                "target_project_id": target_project_id,
            }
        }
```

Reply:
Come to think of it, I think it would be clearer if I definitely got rid of repetitive phrases. thank you : )